bip78/src/params.rs

```diff
@@ -0,0 +1,83 @@
+use std::fmt;
+
+use crate::fee_rate::FeeRate;
+
+pub(crate) struct Params {
+    // version
+    // v: usize,
+    // disableoutputsubstitution
+    pub disable_output_substitution: bool,
+    // maxadditionalfeecontribution, additionalfeeoutputindex
+    pub additional_fee_contribution: Option<(bitcoin::Amount, usize)>,
+    // minfeerate
+    pub min_feerate: FeeRate,
+}
+
+impl Default for Params {
+    fn default() -> Self {
+        Params {
+            disable_output_substitution: false,
+            additional_fee_contribution: None,
+            min_feerate: FeeRate::ZERO,
+        }
+    }
+}
+
+impl Params {
+    #[cfg(feature = "receiver")]
+    pub fn parse(query: &str) -> Result<Params, ParamsError> {
+
+        let mut params = Params::default();
+
+        let mut additional_fee_output_index = None;
+        let mut max_additional_fee_contribution = None;
+
+        let query =  url::Url::parse(query).ok();
+        if let Some(query) = query {
+            for (k, v) in query.query_pairs().into_owned() {
+                match (k.as_str(), v) {
+                    ("v", v) => if v != "1" {
+                        return Err(ParamsError::UnknownVersion)
+                    },
+                    ("additionalfeeoutputindex", index) =>  {
+                        if let Ok(index) = index.parse::<usize>() {
+                            // Check for index out of bounds at fee application.
+                            // Params doesn't need to know about psbt.
+                            additional_fee_output_index = Some(index);
+                        }
+                    },
+                    ("maxadditionalfeecontribution", fee) => {
+                        max_additional_fee_contribution = bitcoin::Amount::from_str_in(&fee, bitcoin::Denomination::Bitcoin).ok();
+                    }
+                    ("minfeerate", feerate) => {
+                        if let Ok(rate) = feerate.parse::<u64>() {
+                            params.min_feerate = FeeRate::from_sat_per_vb(rate);
+                        }
+                    }
+                    ("disableoutputsubstitution", _) => params.disable_output_substitution = true, // existance is truthy
```

Comment:
Ok, we should change it in the sender mod & loptos too which [expect 1](https://github.com/Kixunil/loptos/blob/master/src/main.rs#L296)

I couldn't think of a situation where the param would be deliberately set false, so I took liberty to make a catchall to avoid error

Reply:
Oh, yeah, those are bugs, it seems I got confused before. Also we should probably check other implementations (C#, Electrum) what they do and if they diverge file issues to unify them.

Reply:
done, I check `v == "true"`

Reply:
I thought about this more, and although it conforms to spec, we're breaking [Postel's law](https://lawsofux.com/postels-law/) of UX by conforming so rigidly. We could check for 0, false and ignore, but otherwise accept based on existance

Reply:
I really don't like Postel's law. It makes things very hard to debug and leads to confusing problems. HTML is a prime example of such problems. And I'm not the only one, I've seen people arguing against it. Even [IETF posted a memo arguing against it](https://datatracker.ietf.org/doc/html/draft-thomson-postel-was-wrong-00).

I've spent countless of hours debugging silently ignore errors in various software. It's among the least fun programming tasks I know of. If you know of any implementation that doesn't conform to the specs let's open an issue there. I'd rather send a PR than deal with silent errors. *Especially* in Bitcoin where a silent misinterpretation may mean direct loss of funds.